cpp/search/search.cpp

```diff
@@ -2001,28 +2001,26 @@ void Search::initNodeNNOutput(
       getOpp(thread.pla) == playoutDoublingAdvantagePla ? -searchParams.playoutDoublingAdvantage : searchParams.playoutDoublingAdvantage
     );
   }
-  if(isRoot && searchParams.rootNumSymmetriesToSample > 1) {
-    vector<shared_ptr<NNOutput>> ptrs;
-    for(int i = 0; i<searchParams.rootNumSymmetriesToSample; i++) {
-      bool skipCacheThisIteration = skipCache || i > 0; //Skip cache on subsequent iterations to get new random draws for orientation
-      nnEvaluator->evaluate(
-        thread.board, thread.history, thread.pla,
-        nnInputParams,
-        thread.nnResultBuf, skipCacheThisIteration, includeOwnerMap
-      );
-      ptrs.push_back(std::move(thread.nnResultBuf.result));
-    }
-    node.nnOutput = std::shared_ptr<NNOutput>(new NNOutput(ptrs));
+  // We will use numSymmetriesToSample when conputing the hash so that multiple searchers with different settings won't use a cached
+  // eval that had the wrong numSymmetriesToSample
+  nnInputParams.numSymmetriesToSample = isRoot ? searchParams.rootNumSymmetriesToSample : searchParams.nodeNumSymmetriesToSample;
+  if(nnInputParams.numSymmetriesToSample > 1 || skipCache) {
```

Comment:
skipCache leading to evaluate_symmetry is weird, I would expect no || here and the skipCache for evaluate to handle it

Reply:
That's a good point. Because we don't check skipCache below in the normal evaluate() call. It might be an artifact of the told test against "i > 0". So just check here for >1 and do the if/else and pass skipCache in the function like usual.

In fact, num == 1 and skipCache true would call wrong function. Thanks.